website/docs/reference/widgets/list.md

```diff
@@ -41,69 +35,163 @@ You can display static JSON data in the `Items` property for generating the list
 
 Follow the steps below to paginate the responses and request smaller chunks of data at a time:
 
-1. Enable the Server Side Pagination property for the List.
-2. Call the query on the onPageChange event listener.
-3. Set the LIMIT and OFFSET clause in the query:
+1. Enable the **Server Side Pagination** property for the List.
+2. Call the query on the **onPageChange** event listener.
+3. Set the `LIMIT` and `OFFSET` clauses in the query using the **pageSize** and **pageNo** properties of the List as shown below:
 
 ```javascript
-SELECT * FROM users LIMIT {{ List1.pageSize }} OFFSET {{ (List1.pageNo - 1) * List1.pageSize }}
+SELECT * FROM users LIMIT {{ <listName>.pageSize }} OFFSET {{ (<listName>.pageNo - 1) * <listName>.pageSize }}
 ```
 
 ## Access list items
 
-You can reference the values inside each item selected in the list using the `selectedItem` property as shown below:
+* You can reference the values of each field inside the array dataset bound to the List for the selected list item using the **selectedItem** as shown below:
+
+  ```javascript
+
+  {{<listName>.selectedItem.<fieldName>}}
+  ```
+
+  **Example** - In the preceding example, if you want to display the book name of the selected item in a Text widget, bind it in the **Text** property of the Text widget as shown below 
+
+  ```javascript
+
+  {{<listName>.selectedItem.bookName}}
+  ```
+
+* To access the index of the selected item in the List, you can use the **selectedItemIndex** as shown below:
+
+  ```javascript
+  {{<listName>.selectedItemIndex}}
+  ```
+
+* You can access the sibling widgets inside a list item card using the **currentView** property. 
+
+  **Example** - If you have an Input widget and a Button widget inside the List and want to use the Input's **Text** property to show an alert message on button click. In the button widget's *onClick* event, you can access the input widget's value as shown below
+
+  ```javascript
+  {{showAlert(currentView.<inputName>.text)}}
+  ```
+  The **currentView** property should always be used to access sibling data instead of referencing it directly Eg: `{{Input1.text}` may seem to work in the app's Edit mode but won't work when deployed.   
+
+* The **currentItemView** helps you to view the data and the state of the widgets present in all the items of the List widget and is an *array* of *objects* with each widget's state represented as an object. The *array* of *objects* is limited to the number of items visible on the page rather than the number list items present. If there are 300 objects in the list data, but the List widget is showing 5 items per page, then the **currentItemsView** property shows an *array* of only 5 *objects*.
```

Comment:
```suggestion
* The **currentItemsView** helps you to view the data and the state of the widgets present in all the items of the List widget and is an *array* of *objects* with each widget's state represented as an object. The *array* of *objects* is limited to the number of items visible on the page rather than the number list items present. If there are 300 objects in the list data, but the List widget is showing 5 items per page, then the **currentItemsView** property shows an *array* of only 5 *objects*.
```